UC NetCode Fall 2024/Assets/App/Resource/Scripts/UI/LobbyManager.cs

```diff
@@ -0,0 +1,189 @@
+        }
+        else
+        {
+            rdytxt.text = "Not Ready";
+        }
+        RdyBttnToggleServerRpc(_isReady);
+    }
+
+    [Rpc(target:SendTo.Server, RequireOwnership = false)]
+    private void RdyBttnToggleServerRpc(bool readyStatus, RpcParams rpcParams = default)
+    {
+        Debug.Log(message: "From Rdy bttn RPC");
+        _networkPlayers.UpdateReadyClient(rpcParams.Receive.SenderClientId, readyStatus);
+    }
+
+    private void leaveBttnClick()
+    {
+        if (!IsServer) 
+        {
+            QuitLobbyServerRpc();
+            SceneManager.LoadScene(0);
+
+        }
+        else
+        {
+            foreach (PlayerInfoData playerdata in _networkPlayers._allConnectedPlayers)
+            {
+                if(playerdata._clientId != _myLocalClientId)
+                {
+                    KickUserBttn(playerdata._clientId);
+                }
+            }
+            NetworkManager.Shutdown();
+            SceneManager.LoadScene(0);
+        }
+    }
+
+    [Rpc(target:SendTo.Server, RequireOwnership = false)]
+    private void QuitLobbyServerRpc(RpcParams rpcParams=default)
+    {
+        KickUserBttn(rpcParams.Receive.SenderClientId);
+    }
+
+    private void NetPlayersChanged(NetworkListEvent<PlayerInfoData> changeevent)
+    {
+        Debug.Log(message: "Net Players had changed event fired!");
+        PopulateLabels();
+    }
+
+    [ContextMenu(itemName:"PopulateLabel")]
+    private void PopulateLabels()
+    {
+        ClearPlayerPanel();
+
+        bool allReady = true;
+
+        foreach (PlayerInfoData playerData in _networkPlayers._allConnectedPlayers)
+        {
+            GameObject newPlayerPanel = Instantiate(_panelPrefab, _ContentGo.transform);
+            PlayerLabel _playerLabel = newPlayerPanel.GetComponent<PlayerLabel>();
+
+            _playerLabel.OnKickClicked += KickUserBttn;
+
+            if(IsServer && playerData._clientId != _myLocalClientId)
+            {
+                _playerLabel.setKickActive(isOn: true);
+                _readyBttn.GameObject().SetActive(false);
+            }
+            else
+            {
+                _playerLabel.setKickActive(isOn: false);
+                
+            }
+
+            _playerLabel.SetPlayerLabelName(playerData._clientId);
+            _playerLabel.SetReady(playerData._isPlayerReady);
+            _playerLabel.setPlayerColor(playerData._colorId);
+            _PlayerPanels.Add(newPlayerPanel);
+
+            if (playerData._isPlayerReady == false)
+            {
+                allReady = false;
+            }
+        }
+        if (IsServer)
+        {
+            if (allReady)
+            {
+                if(_networkPlayers._allConnectedPlayers.Count > 1)
+                {
+                    rdytxt.text = "Ready to start";
+                    _startBttn.gameObject.SetActive(true);
+                }
+                else
+                {
+                    rdytxt.text = "Empty Lobby";
```

Comment:
not a game breaking bug, but I noticed that the start menu will appear if you are alone in the lobby. this would be more or less a quality of life change by setting the startbutton game object set active to false at this level.